pkg/cmd/server/kubernetes/master.go

```diff
@@ -160,12 +73,30 @@ func newMasterLeases(storage storage.Interface) election.Leases {
 
 // RunNamespaceController starts the Kubernetes Namespace Manager
 func (c *MasterConfig) RunNamespaceController(kubeClient kclientset.Interface, clientPool dynamic.ClientPool) {
-	// Find the list of namespaced resources via discovery that the namespace controller must manage
-	groupVersionResources, err := kubeClient.Discovery().ServerPreferredNamespacedResources()
+	// TODO: consider using a list-watch + cache here rather than polling
+	var gvrFn func() ([]unversioned.GroupVersionResource, error)
+	rsrcs, err := kubeClient.Discovery().ServerResources()
 	if err != nil {
-		glog.Fatalf("Failed to get supported resources from server: %v", err)
+		glog.Fatalf("Failed to get group version resources: %v", err)
+	}
+	for _, rsrcList := range rsrcs {
```

Comment:
if we're not supporting TPR in origin yet, I'd rather not do this... there are still issues with namespace deletion when a TPR exists and I don't want to inherit those issues for something we don't support.